packages/live-preview-sdk/src/rest/entities.ts

```diff
@@ -33,75 +26,61 @@ function getFieldName(contentType: ContentType, field: ContentType['fields'][num
   return field.apiName || field.name;
 }
 
+function isInDepthLimit(maxDepth: number): boolean {
+  return maxDepth >= 0;
+}
+
+function getRichTextDepth(maxDepth: number) {
+  return Math.min(maxDepth - 1, MAX_RTE_DEPTH);
+}
+
 /**
- * Update the reference from the entry editor with the information from the entityReferenceMap.
+ * Update the reference from the entry editor with the information from the store.
  * If the information is not yet available, it send a message to the editor to retrieve it.
  */
 async function updateRef(
   dataFromPreviewApp: Reference | undefined,
   updateFromEntryEditor: Reference,
   locale: string,
-  entityReferenceMap: EntityReferenceMap,
-  depth: number,
-  visitedReferenceMap: Map<string, Reference>,
-  sendMessage: SendMessage
+  maxDepth: number,
```

Comment:
why have you changed this from depth to maxDepth sorry I think I'm missing this

Reply:
So we can define a different _maxDepth_ for RichText

it's easier in this one, as we otherwise would need to start with `8` for rich text and check for the limit of 10